rust/src/types/id.rs

```diff
@@ -0,0 +1,145 @@
+// Copyright (c) 2025 Sonic Operations Ltd
+//
+// Use of this software is governed by the Business Source License included
+// in the LICENSE file and at soniclabs.com/bsl11.
+//
+// Change Date: 2028-4-16
+//
+// On the date above, in accordance with the Business Source License, use of
+// this software will be governed by the GNU Lesser General Public License v3.
+
+use zerocopy::{FromBytes, Immutable, IntoBytes, Unaligned};
+
+use crate::types::NodeType;
+
+/// An identifier for a node in a file based Verkle trie.
+// NOTE: Changing the layout of this struct will break backwards compatibility of the
+// serialization format.
+#[derive(Debug, Clone, Copy, PartialEq, Eq, Hash, FromBytes, IntoBytes, Immutable, Unaligned)]
+#[repr(transparent)]
+pub struct NodeId([u8; 6]);
+
+impl NodeId {
+    pub const EMPTY_NODE_PREFIX: u64 = 0x0000_0000_0000_0000;
+    pub const INNER_NODE_PREFIX: u64 = 0x0000_4000_0000_0000;
+    pub const LEAF_NODE_2_PREFIX: u64 = 0x0000_8000_0000_0000;
+    pub const LEAF_NODE_256_PREFIX: u64 = 0x0000_C000_0000_0000;
+
+    const PREFIX_MASK: u64 = 0x0000_C000_0000_0000;
+    const INDEX_MASK: u64 = !Self::PREFIX_MASK;
+
+    /// Creates a new [`NodeId`] from a [`u64`] index and a [`NodeType`].
+    /// The index must be smaller than 2^46.
+    pub fn from_idx_and_node_type(idx: u64, node_type: NodeType) -> Self {
```

Comment:
```suggestion
    pub fn from_idx_and_node_type(idx: u64, node_type: NodeType) -> Self {
        assert!(idx < 2^46);
```

Why not check for this precondition?